glib/src/subclass/types.rs

```diff
@@ -115,19 +115,72 @@ pub unsafe trait ClassStruct: Sized + 'static {
 }
 
 /// Trait for subclassable class structs.
-pub unsafe trait IsSubclassable<T: ObjectSubclass>: crate::object::IsClass {
+pub unsafe trait IsSubclassable<T: ObjectSubclass>: IsSubclassableDefault<T> {
     /// Override the virtual methods of this class for the given subclass and do other class
     /// initialization.
     ///
     /// This is automatically called during type initialization and must call `class_init()` of the
     /// parent class.
-    fn class_init(class: &mut crate::Class<Self>);
+    fn class_init(class: &mut crate::Class<Self>) {
+        Self::default_class_init(class);
+    }
 
     /// Instance specific initialization.
     ///
     /// This is automatically called during instance initialization and must call `instance_init()`
     /// of the parent class.
-    fn instance_init(instance: &mut InitializingObject<T>);
+    fn instance_init(instance: &mut InitializingObject<T>) {
+        Self::default_instance_init(instance);
+    }
+}
+
+pub trait IsSubclassableDefault<T: ObjectSubclass>: IsClass {
```

Comment:
In that case, if this is only meant to be called from the default impl, we don't really need to make this `pub`, right?

Reply:
Ideally yes, but Rust wouldn't allow the type bound in the definition of `IsSubclassable` if the trait isn't public.

Reply:
I guess with specialization, `IsSubclassableDefault` could be implemented with empty function bodies for everything that implements `IsSubclassable`, with a specialization to call the parent method when `ParentClassIs` is implemented. Then the `IsSubclassable<T>: IsSubclassableDefault<T>`  bound would be unneeded and the trait could be private.

Too bad specialization doesn't seem like it will be stable particularly soon.

Reply:
Ok thanks for the explanation :) Can you put some FIXME comment about that in the code to remind us to possibly look for a better solution once Rust is ready for that?

Reply:
Added `#[doc(hidden)]` and `FIXME` comment.